File: bubbleSort.py
def print_array(arr):
    for num in arr:
        print(num, end=" ")
    print()

def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def main():
    n = int(input("Enter the number of elements: "))
    arr = []

    print("Enter the elements of the array:")
    for _ in range(n):
        element = int(input())
        arr.append(element)

    print("Array before sorting:")
    print_array(arr)

    bubble_sort(arr)

    print("Array after sorting:")
    print_array(arr)

if __name__ == "__main__":
    main()

File: bubbleSort.js
function printArray(arr) {
    for (let num of arr) {
        process.stdout.write(num + " ");
    }
    process.stdout.write("\n");
}

function bubbleSort(arr) {
    const n = arr.length;
    for (let i = 0; i < n - 1; i++) {
        for (let j = 0; j < n - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question("Enter the number of elements: ", (n) => {
        let arr = [];

        console.log("Enter the elements of the array:");
        rl.on('line', (input) => {
            arr.push(parseInt(input));
            if (arr.length === parseInt(n)) {
                console.log("Array before sorting:");
                printArray(arr);

                bubbleSort(arr);

                console.log("Array after sorting:");
                printArray(arr);
                rl.close();
            }
        });
    });
}

main();
